tests/behave/features/steps/legal_adviser.py

```diff
@@ -0,0 +1,33 @@
+from behave import given, when, then
+
+from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.common.by import By
+
+from nose.tools import assert_greater, assert_in
+
+from features.helpers.fala import get_fala_url
+
+
+@given("I am a user searching for legal advisers")
+def step_user_searching_for_legal_advisers(context):
+    get_fala_url(context, "/")
+
+    WebDriverWait(context.driver, 10).until(EC.presence_of_element_located((By.ID, "content")))
+
+
+@when(u'I search for a legal adviser with postcode "{postcode}" and category "{category}"')
+def step_search_legal_adviser(context, postcode, category):
+    context.driver.find_element_by_xpath("//input[@type='text'][@name='postcode']").send_keys(postcode)
+    context.driver.find_element_by_xpath("//input[@name='categories'][@value='{}']".format(category)).click()
+    context.driver.find_element_by_id("searchButton").click()
+    WebDriverWait(context.driver, 10).until(EC.presence_of_element_located((By.CLASS_NAME, "legal-adviser-results")))
+    assert_in("postcode={}".format(postcode), context.driver.current_url)
+    assert_in("categories={}".format(category), context.driver.current_url)
+
+
+@then(u'I should see a list of legal advisers')
+def step_see_list_of_legal_advisers(context):
+    results_of_the_search = context.driver.find_elements_by_xpath("//div[@class='search-results-list']/ul[@class='org-list']/li")
```

Comment:
We have been doing something similar lately using Cypress and in their documentation they recommend not relying so much on CSS class names for selecting elements since these are likely to change without changing the functionality of the app. 

What they suggest doing instead is adding data attributes (e.g. `data-test="blablabla"`) that you can use just for the tests.

It might be useful here as well because the advice is not specific to the framework.
The link is https://docs.cypress.io/guides/references/best-practices.html#Selecting-Elements

Reply:
I really like this idea and will keep this in mind for future tests